src/scores/probability/brier_impl.py

```diff
@@ -65,3 +72,167 @@ def brier_score(
         check_binary(obs, "obs")
 
     return mse(fcst, obs, reduce_dims=reduce_dims, preserve_dims=preserve_dims, weights=weights)
+
+
+def brier_score_for_ensemble(
+    fcst: XarrayLike,  # type: ignore
+    obs: XarrayLike,  # type: ignore
+    ensemble_member_dim: str,
+    event_thresholds: Union[float, int, Sequence[float]],
```

Comment:
This is an area where type hinting my not be helping us as much. I presume the function could also happily take a double, a long or a Decimal, or a sequence of ints, or any of those things. There is probably a more generic numerical type that could be used here. Please consider if there are more types that should be supported in `event_thresholds`

Reply:
I went looking to see if there was a clear answer. There is not. We could define a standard supported set of numerical types in our own typing.py file and use that. Also, should this support numpy types?

Reply:
It could take a numpy type, but it would have to 1d array and then we would have to add further checks that it was a 1d array. At that point, I thought that it would be better to just use the Sequence.

It should be able to work with Sequence[Int] as well. I have used `numbers.real` to simplify this.

Also, now that we have dropped support for Python 3.9, I think we can use `|` instead of `Union` which I think is cleaner.

What are your thoughts on 1d np.array support? (I didn't add it in the latest commit)

Reply:
I didn't see this before leaving my comment related to how type hints may affect versioning. I'm happy to stick with what you've suggested. I am happy not to include the 1d np.array support, I think it's reasonable. For me, type hinting is often about what's practical (and tested / definitely intended to be supported) rather than covering every possible thing that could work. Yes, moving to using | instead of Union is cleaner. I'm happy if that's now something we can migrate to using! :)

Reply:
I'm very happy with Real | Sequence[Real] - looks great, thanks! :)